content/doc/book/pipeline/docker.adoc

```diff
@@ -82,6 +82,45 @@ v14.15.0
 [Pipeline] }
 ----
 
+=== Keep workspace synchronized
+
+Short: if it is important to keep workspace synchronized with other stages, use `reuseNode true`.
+Otherwise, dockerized stage can be run on any other agent or on the same agent, but in temporary workspace.
+
+By default, for containerized stage, Jenkins does:
+
+* pick any agent,
+* create new empty workspace,
+* clone pipeline code into it,
+* mount this new workspace into container.
+
+If you have multiple Jenkins agents, your containerized stage can be started on any of them.
+
+When `reuseNode` set to `true`: no new workspace will be created, and current workspace from current agent will be mounted into container, and container will be started at the same node, so whole data will be synchronized.
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Build') {
+            agent {
+                docker {
+                    image 'gradle:6.7-jdk11'
+                    // Run the container on the node specified at the top-level of the Pipeline, in the same workspace, rather than on a new node entirely:
+                    reuseNode true
+                }
+            }
+            steps {
+                sh 'gradle --version'
+            }
```

Comment:
```suggestion
        stage('Create file in workspace') {
            agent {
                docker {
                    image 'gradle:6.7-jdk11'
                    // Run the container on the node specified at the top-level of the Pipeline, in the same workspace, rather than on a new node entirely:
                    reuseNode true
                }
            }
            steps {
                sh 'touch sample.txt'
            }
        }
        stage('Read file in workspace') {
            agent {
                docker {
                    image 'gradle:6.7-jdk11'
                    // Run the container on the node specified at the top-level of the Pipeline, in the same workspace, rather than on a new node entirely:
                    reuseNode true
                }
            }
            steps {
                sh 'ls sample.txt' // fails if 'reuseNode' is not enabled
            }
```

Reply:
WDYT about this idea (feel free to adapt based on how you feel it representative of real life use cases)? The goal is to show to the reader a case that would work exclusively when having the reuseNode.

Reply:
It is not clear why do we need two containerized stages with one docker image.

I would like to see more real example, like:
* checkout some Java "Hello world" source in first stage (we don't need container here)
* build sources in next stage (for example, we need JDK or Gradle container here)
* test them in the last stage (here we can use JRE container)

Reply:
Good example, which sounds way more "real" than my proposal :)

Don't forget that the first point (e.g. checkout the code) is done implicitly for each stage that starts on a new workspace when using declarative: it means that you can merge the "checkout" with the "build" in your example.

But still, it make sense: the artefacts generated by the build step could be use by the test step, on another container, without requiring a stash/unstash operation: did I udnerstood your example correctly?